md/bdm-in-rest-api.md

```diff
@@ -0,0 +1,146 @@
+        if (c == null) {
+            return buildResponse(responseBuilder, HttpServletResponse.SC_BAD_REQUEST,"""{"error" : "the parameter c is missing"}""")
+        }
+        
+        // use APIClient to retrieve the CarDAO class:
+        def carDAO = context.apiClient.getDAO(CarDAO.class)
+        
+        def currentModel = "DeLorean"
+        // Fetch the cars that match the search criteria:
+        List<Car> cars = carDAO.findByModel(currentModel, p as int, c as int)
+        
+        // Prepare the Json result:
+        // Do NOT return directly the list "cars", as the entire list of Wheel objects would be fetched by lazy loading.
+        // Instead, ONLY select the fields that are necessary for your business logic:
+        def carModels = [:]
+        for (Car car : cars) {
+            // return the fields "id", "buildYear", "color":
+            carModels.put("id: ${car.persistenceId}", "${car.buildYear} - ${car.color}")
+        }
+        
+        def result = [ "model" : currentModel, "number of cars" : cars.size(), "cars" : carModels ]
+        
+        // Attention: do not use "toPrettyString()" if your response is large, as the page "size" may be uselessly big:
+        return buildResponse(responseBuilder, HttpServletResponse.SC_OK, new JsonBuilder(result).toString())
+    }
+}
+```
+::: warning
+**Practices leading to poor performance**
+
+Since wheel1, wheel2, wheel3, wheel4 are lazy loaded, they are **not retrieved** directly when retrieving a Car.
+The retrieval of related Wheel objects is only performed **when accessing the fields** (via getWheel1(), ...), if necessary.
+
+However, when building the response, the default `JsonBuilder` **implicitly fetches** all lazy loaded fields (it calls all field getters).  
+So, if a large number of Business Data is returned and if you have lazy loaded fields in the returned objects, numerous queries are executed, leading to poor performance.
+
+For example, if you don't follow the code sample above and write something like:
+
+```groovy
+        def currentModel = "DeLorean"
+        // Fetch the cars that match the search criteria:
+        List<Car> cars = carDAO.findByModel(currentModel, p as int, c as int)
+        def result = [ "model" : currentModel,"number of cars" : cars.size(), "cars" : cars ]
+        return buildResponse(responseBuilder, HttpServletResponse.SC_OK, new JsonBuilder(result).toString())
+```
+
+The returned result will contain the fields persistenceId, buildYear and color of each car, allowing you to use these in your application(s).  
+However, assuming you want to retrieve 10 cars of the "Delorean" model, this code will execute a total of **41** "Select" database queries
+* 1 query to get the cars,
+* then 4 queries per car to fetch each one of the *wheel* fields to build the JSON response (so 40 queries).
+  
+In comparison, the code following good practises only performs **a single Select database query**.
+
+:::
+
+## Rest API Response content
```

Comment:
I find this block somewhat weirdly placed, might be better if it was above the warning, for consistency?
Not a critical point though.